# Day as a data editor

I first imported the data and loaded the necessary packages.

I highlighted each of the provable facts in the story and will below try to fact check each one

### Results:

| Correct | Wrong | 50/50|
| --- | --- | --- |
| 23 | 11 | 1 |

In [ ]:
import pandas as pd

In [5]:
df = pd.read_csv('powerplants.csv')
df.head()

,facility_code,Plant_Name,utility,util_id,sector_name,city,county,State_Name,zip,street_address,...,solar_capacity,wind_capacity,geothermal_capacity,other_capacity,source_description,tech_type,source_db,report_period,latitude,longitude
0,2,Bankhead Dam,Alabama Power Co,195,Electric Utility,Northport,Tuscaloosa,Alabama,35476,19001 Lock 17 Road,...,0.0,0.0,NaN,0.0,Hydroelectric = 56 MW,Conventional Hydroelectric,"EIA-860, EIA-860M and EIA-923",201801,33.458665,-87.356823
1,3,Barry,Alabama Power Co,195,Electric Utility,Bucks,Mobile,Alabama,36512,North Highway 43,...,0.0,0.0,NaN,0.0,"Coal = 1100.5 MW, Natural Gas = 1072 MW",Conventional Steam Coal; Natural Gas Fired Com...,"EIA-860, EIA-860M and EIA-923",201801,31.006900,-88.010300
2,4,Walter Bouldin Dam,Alabama Power Co,195,Electric Utility,Wetumpka,Elmore,Alabama,36092,750 Bouldin Dam Road,...,0.0,0.0,NaN,0.0,Hydroelectric = 225.9 MW,Conventional Hydroelectric,"EIA-860, EIA-860M and EIA-923",201801,32.583889,-86.283056
3,7,Gadsden,Alabama Power Co,195,Electric Utility,Gadsden,Etowah,Alabama,35903,1000 Goodyear Ave,...,0.0,0.0,NaN,0.0,Natural Gas = 130 MW,Natural Gas Steam Turbine,"EIA-860, EIA-860M and EIA-923",201801,34.012800,-85.970800
4,8,Gorgas,Alabama Power Co,195,Electric Utility,Parrish,Walker,Alabama,35580,460 Gorgas Road,...,0.0,0.0,NaN,0.0,Coal = 1053.7 MW,Conventional Steam Coal,"EIA-860, EIA-860M and EIA-923",201801,33.644344,-87.196486


<hr>

### Sentence: **"The United States has over 8,600 power plants"**


In [32]:
df.shape

(8602, 30)

Verdict: True ✅

<hr>

### Sentence: The us has plants **"across every state"**

In [11]:
df_states = df['State_Name'].value_counts()
df_states.shape

(50,)

Verdict: True ✅

<hr>

### Sentence: The US has plants across **"nearly every county in the nation"**

I checked a few different sources to get an estimate of how many counties there are in the US. I'm going to assume that the total number is about 3100 (roughly in the middle of the three figures)

3143 - [Census Bureau](https://www.census.gov/newsroom/press-releases/2026/2025-popest-metro-micro-counties.html)

3069 - [National Association of Counties](https://www.naco.org/page/what-are-counties)

3244 'counties and county equivalents' or 3007 counties - [Wikipedia](https://en.wikipedia.org/wiki/List_of_United_States_counties_and_county_equivalents)

In [31]:
df_counties = df.groupby(['State_Name', 'county'])['county'].count()
df_counties.shape

(1976,)

Verdict: False ❌

<hr>

### Sentence: **"Gas-fired plants account for about 54 percent of the nation's installed capacity, or roughly 554,000 megawatts"**

In [57]:
df_power_types = pd.melt(df,
                         value_vars=['coal_capacity', 'gas_capacity', 'oil_capacity', 'biomass_capacity', 'hydro_capacity', 'pumped_hydro_cap', 'nuclear_capacity', 'solar_capacity', 'wind_capacity', 'geothermal_capacity', 'other_capacity'],
                         value_name='count')

df_power_types = pd.DataFrame(df_power_types.groupby('variable')['count'].sum())

total_capacity = df_power_types['count'].sum()
df_power_types['pct_total'] = df_power_types['count']/total_capacity
df_power_types.sort_values('count', ascending=False)

,count,pct_total
variable,,
gas_capacity,453396.9,0.418954
coal_capacity,255533.3,0.236122
nuclear_capacity,99629.8,0.092061
wind_capacity,88366.6,0.081654
hydro_capacity,80049.1,0.073968
oil_capacity,33675.5,0.031117
solar_capacity,27442.6,0.025358
pumped_hydro_cap,22808.7,0.021076
biomass_capacity,14165.0,0.013089


Verdict: False ❌

<hr>

### Sentence: coal **"represents nearly a quarter of total capacity across 337 plants producing about 258,000 megawatts"**

In [63]:
df_power_types

,count,pct_total
variable,,
biomass_capacity,14165.0,0.013089
coal_capacity,255533.3,0.236122
gas_capacity,453396.9,0.418954
geothermal_capacity,2508.6,0.002318
hydro_capacity,80049.1,0.073968
nuclear_capacity,99629.8,0.092061
oil_capacity,33675.5,0.031117
other_capacity,4634.7,0.004283
pumped_hydro_cap,22808.7,0.021076


In [117]:
df_coal_plants = df[df['coal_capacity']>0]
df_coal_plants['Plant_Name'].value_counts().shape

(362,)

Verdict: True ✅

<hr>

### Sentence: **"Together, fossil fuels -- natural gas, coal and petroleum -- make up about 68 percent of installed capacity nationwide"**

In [75]:
df_power_types.sort_values('count', ascending=False)

,count,pct_total
variable,,
gas_capacity,453396.9,0.418954
coal_capacity,255533.3,0.236122
nuclear_capacity,99629.8,0.092061
wind_capacity,88366.6,0.081654
hydro_capacity,80049.1,0.073968
oil_capacity,33675.5,0.031117
solar_capacity,27442.6,0.025358
pumped_hydro_cap,22808.7,0.021076
biomass_capacity,14165.0,0.013089


In [77]:
# gas       coal        oil
0.418954 + 0.236122 + 0.031117

0.6861929999999999

Verdict: True ✅

<hr>

### Sentence: **"Nuclear energy, generated by just 45 plants, adds another 22 percent."**

In [79]:
df_power_types.sort_values('count', ascending=False)

,count,pct_total
variable,,
gas_capacity,453396.9,0.418954
coal_capacity,255533.3,0.236122
nuclear_capacity,99629.8,0.092061
wind_capacity,88366.6,0.081654
hydro_capacity,80049.1,0.073968
oil_capacity,33675.5,0.031117
solar_capacity,27442.6,0.025358
pumped_hydro_cap,22808.7,0.021076
biomass_capacity,14165.0,0.013089


In [116]:
df_nuclear_plants = df[df['nuclear_capacity']>0]
df_nuclear_plants['Plant_Name'].value_counts().shape

(61,)

Verdict: False ❌

<hr>

### Sentence: **"Solar leads the country in sheer number of facilities, with 2,247 plants, more than any other single source"**

In [115]:
df_solar_plants = df[df['solar_capacity']>0]
df_solar_plants['Plant_Name'].value_counts().shape

(2280,)

Verdict: True ✅

<hr>

This is an interlude in which I wonder whether all this work I'm putting in will be checked by a human or by AI
<br>And whether this assignment was made by a human or AI...

<hr>

### Sentence: **"Wind farms number over a thousand."**

In [112]:
df_wind_plants = df[df['wind_capacity']>0]
df_wind_plants['Plant_Name'].value_counts().shape

(1054,)

Verdict: True ✅

<hr>

### Sentence: **"But those 2,247 solar installations produce just 27,400 megawatts combined -- roughly 2.5 percent of the national total"**

In [86]:
df_power_types.sort_values('count', ascending=False)

,count,pct_total
variable,,
gas_capacity,453396.9,0.418954
coal_capacity,255533.3,0.236122
nuclear_capacity,99629.8,0.092061
wind_capacity,88366.6,0.081654
hydro_capacity,80049.1,0.073968
oil_capacity,33675.5,0.031117
solar_capacity,27442.6,0.025358
pumped_hydro_cap,22808.7,0.021076
biomass_capacity,14165.0,0.013089


Verdict: True ✅

<hr>

### Sentence: **"The average solar plant clocks in at about 12 megawatts"**

In [89]:
df_solar_plants['solar_capacity'].mean()

np.float64(12.015148861646235)

In [91]:
df_solar_plants['solar_capacity'].median()

np.float64(3.5)

The Median power plant does not clock in at about 12 megawatts, but the mean power plant does. I would just specify which average is being used (I think median might be better in this case)

<hr>

### Sentence: **"By comparison, the average nuclear plant produces over 2,100 megawatts"**

In [92]:
df_nuclear_plants['nuclear_capacity'].mean()

np.float64(1633.2754098360658)

In [93]:
df_nuclear_plants['nuclear_capacity'].median()

np.float64(1707.8)

Verdict: False ❌

<hr>

### Sentence: **"Taken together, all renewable sources -- solar, wind, hydro, geothermal and biomass -- account for about 4,388 facilities"**

In [118]:
df_renewable_plants = df[(df['solar_capacity']>0) |
                         (df['wind_capacity'] > 0) |
                         (df['hydro_capacity'] > 0) |
                         (df['pumped_hydro_cap'] > 0) |
                         (df['geothermal_capacity'] > 0) |
                         (df['biomass_capacity'] > 0)]

df_renewable_plants['Plant_Name'].value_counts().shape

(5595,)

Verdict: False ❌

<hr>

### Sentence: Renewable sources account for **"nearly 30 percent of installed capacity"**

In [121]:
df_power_types.sort_values('count', ascending=False)

,count,pct_total
variable,,
gas_capacity,453396.9,0.418954
coal_capacity,255533.3,0.236122
nuclear_capacity,99629.8,0.092061
wind_capacity,88366.6,0.081654
hydro_capacity,80049.1,0.073968
oil_capacity,33675.5,0.031117
solar_capacity,27442.6,0.025358
pumped_hydro_cap,22808.7,0.021076
biomass_capacity,14165.0,0.013089


In [122]:
## Solar   Wind       Hydro   Pumped hydro  geothermal  biomass
0.025358 + 0.081654 + 0.073968 + 0.021076 + 0.002318 + 0.013089

0.217463

Verdict: False ❌

<hr>

### Sentence: Renewable sources **"outnumber fossil fuel plants nearly two to one, yet still trail far behind in actual power output"**

In [124]:
df_renewable_plants['Plant_Name'].value_counts().shape

(5595,)

In [125]:
df['Plant_Name'].value_counts().shape

(8592,)

In [ ]:
#all plants      renewable plants
8592         -      5595

2997

In [129]:
#renewable      non-renewable plants
5595         /     2997

1.8668668668668669

In [131]:
df_power_types.sort_values('count', ascending=False)

,count,pct_total
variable,,
gas_capacity,453396.9,0.418954
coal_capacity,255533.3,0.236122
nuclear_capacity,99629.8,0.092061
wind_capacity,88366.6,0.081654
hydro_capacity,80049.1,0.073968
oil_capacity,33675.5,0.031117
solar_capacity,27442.6,0.025358
pumped_hydro_cap,22808.7,0.021076
biomass_capacity,14165.0,0.013089


In [ ]:
df_power_types['count'].sum()

np.float64(1082210.8)

In [ ]:
# Total power     wind       hydro     solar     pumped hydro  biomass   geothermal    total power
(1082210.8       - 88366.6 - 80049.1 - 27442.6 - 22808.7     - 14165.0 - 2508.6)    / 1082210.8

0.782537191460296

Verdict: True ✅

<hr>

### Sentence: **"No state has more installed generating capacity than Texas"**

In [139]:
df_power_types_long = pd.melt(df,
                              id_vars=['Plant_Name', 'State_Name'],
                              value_vars=['coal_capacity', 'gas_capacity', 'oil_capacity', 'biomass_capacity', 'hydro_capacity', 'pumped_hydro_cap', 'nuclear_capacity', 'solar_capacity', 'wind_capacity', 'geothermal_capacity', 'other_capacity'],
                              var_name='power_type',
                              value_name='capacity')

df_power_types_long = df_power_types_long[df_power_types_long['capacity']>0]

df_power_types_long

,Plant_Name,State_Name,power_type,capacity
1,Barry,Alabama,coal_capacity,1100.5
4,Gorgas,Alabama,coal_capacity,1053.7
18,E C Gaston,Alabama,coal_capacity,832.0
26,Dolet Hills,Louisiana,coal_capacity,641.0
30,Charles R Lowman,Alabama,coal_capacity,556.0
...,...,...,...,...
94342,Pima Energy Storage System,Arizona,other_capacity,10.0
94442,NRG Elbow Creek Energy Storage Project,Texas,other_capacity,2.0
94489,Panasonic Carport Solar,Colorado,other_capacity,1.0
94509,Hopewell Valley High School,New Jersey,other_capacity,1.0


In [143]:
df_state_capacity = df_power_types_long.groupby('State_Name').sum('capacity')
df_state_capacity.sort_values('capacity', ascending=False).head(5)

,capacity
State_Name,
Texas,120730.5
California,76722.2
Florida,57987.4
Illinois,45199.4
Pennsylvania,44115.6


Verdict: True ✅

<hr>

### Sentence: Texas **"leads the nation at roughly 120,000 megawatts across 323 plants"**

In [146]:
df_state_capacity.sort_values('capacity', ascending=False).head(3)

,capacity
State_Name,
Texas,120730.5
California,76722.2
Florida,57987.4


In [153]:
df_state_plants = df.groupby('State_Name')['Plant_Name'].count()
df_state_plants

State_Name
Alabama             76
Alaska             145
Arizona            129
Arkansas            56
California        1340
Colorado           179
Connecticut         99
Delaware            26
Florida            160
Georgia            159
Hawaii              53
Idaho              138
Illinois           203
Indiana            152
Iowa               240
Kansas             139
Kentucky            51
Louisiana           82
Maine              103
Maryland            99
Massachusetts      348
Michigan           231
Minnesota          364
Mississippi         45
Missouri           125
Montana             57
Nebraska           104
Nevada              81
New Hampshire       61
New Jersey         262
New Mexico         102
New York           407
North Carolina     626
North Dakota        55
Ohio               167
Oklahoma           108
Oregon             173
Pennsylvania       215
Rhode Island        34
South Carolina     111
South Dakota        34
Tennessee           66
Texas              431


Verdict: True ✅

<hr>

### Sentence: **"Natural gas makes up 57 percent of Texas's capacity, followed by wind at 18 percent and coal at just 18 percent"**

In [ ]:
df_texas = df_power_types_long[df_power_types_long['State_Name']=="Texas"]

df_texas_types = df_texas.groupby('power_type')['capacity'].sum()
df_texas_types.sum()

np.float64(120730.5)

In [165]:
df_texas_types

power_type
biomass_capacity      424.9
coal_capacity       20524.0
gas_capacity        69696.6
hydro_capacity        670.0
nuclear_capacity     4960.0
oil_capacity          108.7
other_capacity        564.7
solar_capacity       1221.7
wind_capacity       22559.9
Name: capacity, dtype: float64

In [166]:
# gas     total
69696.6 / 120730.5

0.5772907426043958

In [167]:
# wind     total
22559.9 / 120730.5

0.18686164639424174

In [168]:
# coal     total
20524.0 / 120730.5

0.1699984676614443

Verdict: True ✅

<hr>

### Sentence: **"Texas has more installed wind capacity than any other state, at about 21,900 megawatts"**

In [177]:
df_wind_by_state = df_power_types_long[df_power_types_long['power_type']=='wind_capacity'].groupby('State_Name').sum('capacity')

df_wind_by_state.sort_values('capacity', ascending=False).head(1)

,capacity
State_Name,
Texas,22559.9


Verdict: True ✅

<hr>

### Sentence: **"Kansas ranks second, with about 10,500 megawatts of wind capacity"**

In [176]:
df_wind_by_state.sort_values('capacity', ascending=False).head(2)

,capacity
State_Name,
Texas,22559.9
Oklahoma,7494.8


Verdict: False ❌

<hr>

### Sentence: **"California has the most plants of any state -- 1,340"**

In [182]:
df_state_plants

State_Name
Alabama             76
Alaska             145
Arizona            129
Arkansas            56
California        1340
Colorado           179
Connecticut         99
Delaware            26
Florida            160
Georgia            159
Hawaii              53
Idaho              138
Illinois           203
Indiana            152
Iowa               240
Kansas             139
Kentucky            51
Louisiana           82
Maine              103
Maryland            99
Massachusetts      348
Michigan           231
Minnesota          364
Mississippi         45
Missouri           125
Montana             57
Nebraska           104
Nevada              81
New Hampshire       61
New Jersey         262
New Mexico         102
New York           407
North Carolina     626
North Dakota        55
Ohio               167
Oklahoma           108
Oregon             173
Pennsylvania       215
Rhode Island        34
South Carolina     111
South Dakota        34
Tennessee           66
Texas              431


Verdict: True ✅

<hr>

This is an interlude in which I wonder whether all this work I'm putting in will be checked by a human or by AI
<br>And whether this assignment was made by a human or AI...

<hr>

### Sentence: California **"ranks second in total capacity at about 77,000 megawatts"**

In [180]:
df_state_capacity.sort_values('capacity', ascending=False).head(2)

,capacity
State_Name,
Texas,120730.5
California,76722.2


Verdict: True ✅

<hr>

### Sentence: **"half of California's capacity comes from renewable sources"**

In [187]:
df_california = df_power_types_long[df_power_types_long['State_Name']=="California"]

df_california_types = df_california.groupby('power_type')['capacity'].sum()
df_california_types.sum()

np.float64(76722.2)

In [188]:
df_california_types

power_type
biomass_capacity        1297.9
coal_capacity             57.0
gas_capacity           39393.9
geothermal_capacity     1865.8
hydro_capacity         10197.7
nuclear_capacity        2240.0
oil_capacity             449.7
other_capacity           495.1
pumped_hydro_cap        3911.9
solar_capacity         10962.0
wind_capacity           5851.2
Name: capacity, dtype: float64

In [189]:
#biomass    geothermal  hydro     pumped hydro  solar     wind      total
(1297.9 +   1865.8 +    10197.7 +   3911.9 +    10962 +   5851.2) / 76722.2

0.44428470507884293

Verdict: True ✅

<hr>

### Sentence: California's 50% renewable energy is the **"highest share among the ten largest energy-producing states"**

In [210]:
df.dtypes

facility_code            int64
Plant_Name              object
utility                 object
util_id                  int64
sector_name             object
city                    object
county                  object
State_Name              object
zip                      int64
street_address          object
Fuel_Type               object
Installed_MW           float64
Total_MW               float64
coal_capacity          float64
gas_capacity           float64
oil_capacity           float64
biomass_capacity       float64
hydro_capacity         float64
pumped_hydro_cap       float64
nuclear_capacity       float64
solar_capacity         float64
wind_capacity          float64
geothermal_capacity    float64
other_capacity         float64
source_description      object
tech_type               object
source_db               object
report_period            int64
latitude               float64
longitude              float64
dtype: object

In [215]:
all_capacity_columns = ['coal_capacity', 'gas_capacity', 'oil_capacity', 'biomass_capacity', 'hydro_capacity', 'pumped_hydro_cap', 'nuclear_capacity', 'solar_capacity', 'wind_capacity', 'geothermal_capacity', 'other_capacity']

renewable_capacity_columns = ['biomass_capacity', 'hydro_capacity', 'pumped_hydro_cap', 'solar_capacity', 'wind_capacity', 'geothermal_capacity']

df_by_states = df.groupby('State_Name').sum()
df_by_states['capacity_of_renewables'] = df_by_states[renewable_capacity_columns].sum(axis=1, skipna=True)
df_by_states['total_capacity'] = df_by_states[all_capacity_columns].sum(axis=1, skipna=True)
df_by_states['pct_renewable'] = df_by_states['capacity_of_renewables']/df_by_states['total_capacity']

df_by_states.head()

,facility_code,Plant_Name,utility,util_id,sector_name,city,county,zip,street_address,Fuel_Type,...,other_capacity,source_description,tech_type,source_db,report_period,latitude,longitude,capacity_of_renewables,total_capacity,pct_renewable
State_Name,,,,,,,,,,,,,,,,,,,,,
Alabama,1840707,Bankhead DamBarryWalter Bouldin DamGadsdenGorg...,Alabama Power CoAlabama Power CoAlabama Power ...,1264336,Electric UtilityElectric UtilityElectric Utili...,NorthportBucksWetumpkaGadsdenParrishDemopolisO...,TuscaloosaMobileElmoreEtowahWalkerGreeneCalhou...,2744604,19001 Lock 17 RoadNorth Highway 43750 Bouldin ...,hydroelectricnatural gashydroelectricnatural g...,...,19.8,"Hydroelectric = 56 MWCoal = 1100.5 MW, Natural...",Conventional HydroelectricConventional Steam C...,"EIA-860, EIA-860M and EIA-923EIA-860, EIA-860M...",15336876,2489.386514,-6590.165346,4170.9,29522.6,0.141278
Alaska,3527170,Annex CreekGold CreekLemon CreekSalmon Creek 1...,Alaska Electric Light&Power CoAlaska Electric ...,1643326,Electric UtilityElectric UtilityElectric Utili...,JuneauJuneauJuneauJuneauSkagwayHainesKetchikan...,JuneauJuneauJuneauJuneauSkagway Hoonah AngoonH...,14458786,Annex Creek700 Capital Ave.5601 Tonsgard Ct.27...,hydroelectricpetroleumpetroleumhydroelectricpe...,...,45.0,"Hydroelectric = 3.6 MWHydroelectric = 1.6 MW, ...",Conventional HydroelectricConventional Hydroel...,"EIA-860, EIA-860M and EIA-923EIA-860, EIA-860M...",29261145,8853.702766,-21693.128332,538.0,2742.2,0.196193
Arizona,5897001,South ConsolidatedChollaDouglasOcotilloWest Ph...,Salt River ProjectArizona Public Service CoAri...,4685042,Electric UtilityElectric UtilityElectric Utili...,PhoenixJoseph CityDouglasTempePhoenixRed RockY...,MaricopaNavajoCochiseMaricopaMaricopaPinalYuma...,11010471,P.O. Box 52025 MS PAB2214801 Cholla Lake RoadS...,hydroelectriccoalpetroleumnatural gasnatural g...,...,40.0,Hydroelectric = 1.4 MWCoal = 767 MWPetroleum =...,Conventional HydroelectricConventional Steam C...,"EIA-860, EIA-860M and EIA-923EIA-860, EIA-860M...",26032329,4298.394163,-14441.781205,5234.4,28618.7,0.182901
Arkansas,1532477,CarpenterLake CatherineOsceola (AR)RemmelBeave...,Entergy Arkansas IncEntergy Arkansas IncCity o...,1118466,Electric UtilityElectric UtilityElectric Utili...,Hot SpringsJones MillOsceolaMalvernEureka Spri...,GarlandHot SpringMississippiHot SpringCarrollC...,4048113,"1398 Carpenter Dam RoadHwy. 270, PO Box 218303...",hydroelectricnatural gaspetroleumhydroelectric...,...,0.0,Hydroelectric = 62 MWNatural Gas = 519.7 MWPet...,Conventional HydroelectricNatural Gas Steam Tu...,"EIA-860, EIA-860M and EIA-923EIA-860, EIA-860M...",11300856,1962.562004,-5180.020411,1719.1,14782.1,0.116296
California,58050304,RollinsVenice HydroJ S EastwoodMcClureTurlock ...,Nevada Irrigation DistrictMetropolitan Water D...,50848004,IPP Non-CHPElectric UtilityElectric UtilityEle...,ColfaxCulver CityShaver LakeModestoHickmanHick...,PlacerLos AngelesFresnoStanislausStanislausSta...,124032613,15531 Arrow Head Lane3815 Sepulveda BlvdHwy 16...,hydroelectrichydroelectricpumped storagenatura...,...,495.1,Hydroelectric = 12.1 MWHydroelectric = 10.1 MW...,Conventional HydroelectricConventional Hydroel...,"EIA-860, EIA-860M and EIA-923EIA-860, EIA-860M...",270413340,48209.995465,-159765.950091,34086.5,76722.2,0.444285


In [219]:
df_renewable_share = df_by_states[['capacity_of_renewables', 'total_capacity', 'pct_renewable']]
df_renewable_share.sort_values('total_capacity', ascending=False).head(10)

,capacity_of_renewables,total_capacity,pct_renewable
State_Name,,,
Texas,24876.5,120730.5,0.206050
California,34086.5,76722.2,0.444285
Florida,2142.0,57987.4,0.036939
Illinois,4447.9,45199.4,0.098406
Pennsylvania,4462.7,44115.6,0.101159
New York,8625.7,40249.9,0.214304
Georgia,5836.3,36880.7,0.158248
North Carolina,6117.5,32934.4,0.185748
Washington,25116.8,31061.0,0.808628


Verdict: False ❌

<hr>

### Sentence: **"Florida, with nearly 58,000 megawatts of installed capacity"**

In [191]:
df_florida = df_power_types_long[df_power_types_long['State_Name']=="Florida"]
df_florida['capacity'].sum()

np.float64(57987.4)

Verdict: True ✅

<hr>

### Sentence: Florida **"gets just 3.7 percent from renewables"**

In [192]:
df_florida_types = df_florida.groupby('power_type')['capacity'].sum()
df_florida_types.sum()

np.float64(57987.399999999994)

In [193]:
df_florida_types

power_type
biomass_capacity     1279.2
coal_capacity        8629.0
gas_capacity        38325.7
hydro_capacity         54.5
nuclear_capacity     3572.0
oil_capacity         4970.0
other_capacity        348.7
solar_capacity        808.3
Name: capacity, dtype: float64

In [194]:
#biomass    hydro    solar    total
(1279.2 +   54.5 +   808.3) / 57987.39

0.03693906554511248

Verdict: True ✅

<hr>

### Sentence: **"Mississippi sits at 2.7 percent"** renewable energy

In [224]:
df_renewable_share.sort_values('pct_renewable').head(2)

,capacity_of_renewables,total_capacity,pct_renewable
State_Name,,,
Delaware,46.4,3377.8,0.013737
Mississippi,435.3,16114.4,0.027013


Verdict: True ✅

<hr>

### Sentence: **"Ohio and Louisiana are both below 4.0 percent"** renewable energy

In [225]:
df_renewable_share.sort_values('pct_renewable').head(4)

,capacity_of_renewables,total_capacity,pct_renewable
State_Name,,,
Delaware,46.4,3377.8,0.013737
Mississippi,435.3,16114.4,0.027013
Louisiana,683.2,23743.7,0.028774
Ohio,923.4,30050.4,0.030728


Verdict: True ✅

<hr>

### Sentence: **"Grand Coulee Dam -- the single largest power plant in the country at 7,079 megawatts -- sits in Oregon"**

In [231]:
df.sort_values('Total_MW', ascending=False)[['Plant_Name', 'Total_MW']].head(1)

,Plant_Name,Total_MW
1897,Grand Coulee,7079.0


Verdict: True ✅

<hr>

### Sentence: Oregon **"leads large states with 74 percent of its capacity from renewable sources"**

In [236]:
df_renewable_share.sort_values('pct_renewable', ascending=False).head(5)

,capacity_of_renewables,total_capacity,pct_renewable
State_Name,,,
Vermont,636.4,736.3,0.864322
Washington,25116.8,31061.0,0.808628
Idaho,4013.3,5165.8,0.776898
Oregon,12188.0,16511.1,0.738170
South Dakota,2436.8,4128.6,0.590224


Verdict: False ❌

<hr>

### Sentence: **"Washington follows close behind"** Oregan in renewable energy share **"at 81 percent"**

In [237]:
df_renewable_share.sort_values('pct_renewable', ascending=False).head(5)

,capacity_of_renewables,total_capacity,pct_renewable
State_Name,,,
Vermont,636.4,736.3,0.864322
Washington,25116.8,31061.0,0.808628
Idaho,4013.3,5165.8,0.776898
Oregon,12188.0,16511.1,0.738170
South Dakota,2436.8,4128.6,0.590224


Verdict: False ❌

<hr>

### Sentence: **"The median power plant in this country generates just 95 megawatts"**

In [238]:
df['Total_MW'].median()

np.float64(10.0)

Verdict: False ❌

<hr>

### Sentence: **"the ten largest plants in the country -- a mix of hydro, nuclear, gas and coal -- each exceed 3,000 megawatts"**

In [242]:
df_power_types_long.sort_values('capacity', ascending=False).head(10)

,Plant_Name,State_Name,power_type,capacity
36305,Grand Coulee,Washington,hydro_capacity,6765.0
53406,Palo Verde,Arizona,nuclear_capacity,3937.0
13170,West County Energy Center,Florida,gas_capacity,3669.0
1941,Scherer,Georgia,coal_capacity,3392.0
51634,Browns Ferry,Alabama,nuclear_capacity,3309.4
412,Bowen,Georgia,coal_capacity,3232.0
1863,Gibson,Indiana,coal_capacity,3132.0
835,Monroe (MI),Michigan,coal_capacity,3066.0
44910,Bath County,Virginia,pumped_hydro_cap,3003.0
1680,John E Amos,West Virginia,coal_capacity,2900.0


Verdict: True ✅

<hr>

### Sentence: **"The top ten alone account for roughly 39,000 megawatts, or about 3.6 percent of the national total"**

In [247]:
df_top_10_plants = df_power_types_long.sort_values('capacity', ascending=False).head(10)
df_top_10_plants['capacity'].sum()

np.float64(36405.4)

In [249]:
df_power_types_long['capacity'].sum()

np.float64(1082210.8)

In [251]:
#top 10     all plants
36405.4 /   1082210.8

0.033639841701820015

Verdict: True ✅

<hr>

### Sentence: 39,000 megawatts **"is less capacity than the state of Florida, which has 58,000 megawatts**"

In [253]:
df_florida['capacity'].sum()

np.float64(57987.4)

Verdict: True ✅

<hr>